src/control_packet.rs

```diff
@@ -11,14 +11,17 @@ bitfield! {
     /// Is the packet a request?
     pub rq, set_rq : 0, 0;
     /// Is this packet a datagram?
-    d, set_d: 1, 1;
-    rsvd, _: 2, 2;
-    instance_id, set_instance_id: 7, 3;
+    pub d, set_d: 1, 1;
+    /// Reserved bits
+    pub rsvd, _: 2, 2;
```

Comment:
We probably don't need the reserved bit to be public do we?

Reply:
Good point, I've changed that.